tests/implementations/selector.test.ts

```diff
@@ -0,0 +1,175 @@
+import { Selector as SelectorInterface } from '../../src/api/Selector';
+import { avengers, beatles, Character, SimpleCharacterKey } from '../mocks/items';
+import { Selector } from '../../src';
+import { errorMessages, validationMessages } from '../../src/helpers/messages';
+
+describe('Selector service test suite', () => {
+  let selector: SelectorInterface<Character, SimpleCharacterKey>;
+
+  beforeEach(() => {
+    selector = new Selector<Character, SimpleCharacterKey>();
+  });
+
+  it('Call setItems with items and get items at subscription after one emission', async (done) => {
+    expect.assertions(1);
+    await selector.setItems({ items: beatles });
+    selector.items$({}).subscribe((items: any) => {
```

Comment:
please fix all **any** (I will not add more comments about it, but there are some)

Why do you need done here? If you put the amount of expectations - test should wait automatically